php-classes/Emergence/Connectors/AbstractSpreadsheetConnector.php

```diff
@@ -192,7 +193,14 @@ protected static function _saveRecord(IJob $Job, ActiveRecord $Record, $pretend,
         } elseif ($logEntry['action'] == 'update') {
             $results['updated']++;
 
-            foreach (array_keys($logEntry['changes']) AS $changedField) {
+            $changes = [];
+            if (is_a($logEntry['changes'], KeyedDiff::class)) {
+                $changes = array_keys($logEntry['changes']->getNewValues());
+            } elseif (is_array($logEntry['changes'])) { // legacy support
+                $changes = array_keys($logEntry['changes']);
+            }
+
+            foreach ($changes AS $changedField) {
```

Comment:
I _think_ we can just assume `changes` is a `KeyedDiff` instance and loop through it directly using its interface. I tried maintaining some support of both and ended up instead opting for the hard turn to using KeyedDiff universally

In this case I think that's a safe strategy, as `$logEntry['changes']` here can only come out of `$Job->logRecordDelta` and there's only one implementation of that, and it uses `KeyedDiff`